onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -465,12 +465,7 @@ def _create_inference_session(self, providers, provider_options, disabled_optimi
         )
         if not providers and len(available_providers) > 1:
             self.disable_fallback()
-            raise ValueError(
-                f"This ORT build has {available_providers} enabled. "
-                "Since ORT 1.9, you are required to explicitly set "
-                "the providers parameter when instantiating InferenceSession. For example, "
-                f"onnxruntime.InferenceSession(..., providers={available_providers}, ...)"
-            )
+            warnings.warn(f"This ORT build has {available_providers} enabled, use CPUExecutionProvider as default")
```

Comment:
wait. why? We do want to enforce explicit EP registration for non-default providers.

Reply:
discussed offline. now seems the right time to remove this check completely. it's been many releases since ORT 1.9

Reply:
Discussed with George. Let's remove this restriction altogether.

Reply:
The warning is not issued by other language bindings. If we want to remain consistent, we should either remove this altogether or add the warning for all bindings. Python is no special.

Reply:
warning isn't strictly needed.